R/CrmPackClass-methods.R

```diff
@@ -0,0 +1,52 @@
+# tidy ----
+
+## generic ----
+
+#' Tidying `CrmPackClass` objects
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' In the spirit of the `broom` package, provide a method to convert a
+#' `CrmPackClass` object to a (list of) tibble9s).
+#'
+#' @param x (`CrmPackClass`)\cr the object to be tidied.
+#' @param ... \cr Used by class-specific methods
+#'
+#' @return A (list of) tibble(s) representing the object in tidy form.
+#'
+#' @export
+#'
+setGeneric(
+  name = "tidy",
+  def = function(x, ...) {
+    standardGeneric("tidy")
+  }
+)
+
+## CrmPackClass ----
+
+#' Tidy a `CrmPackClass` Object
+#'
+#' Following the principles of the `broom` package, convert a `CrmPackClass`
+#' object to a (list of) tibbles.  This is a basic, default representation
+#'
+#' @param x (`CrmPackClass`)\cr The object to be tidied
+#' @param ... \cr Not used at present
+#' @rdname tidy
+#' @aliases tidy-CrmPackClass
+#' @keywords methods
+#' @example examples/CrmPackClass-method-tidy.R
+#'
+#' @export
+setMethod(
+  f = "tidy",
+  signature = signature(x = "CrmPackClass"),
+  definition = function(x, ...) {
+    rv <- h_tidy_all_slots(x) |> h_tidy_class(x)
```

Comment:
general comment: it would be preferable to avoid the pipe operator in package code. reason being that it is much easier to debug line by line when it is not just a single call being evaluated but multiple ones. plus usually it is better readable when you have interim variables with nice "speaking" names.
but I see that you have a lot of code now with the pipe operator. so leave it up to you whether you want to change that

Reply:
I think this is a matter of style.  To be honest, I find it much easier to read pipes than "conventional" R code.  But I agree it's a personal preference.
The conventional equivalent of this line is 
```
rv <-  h_tidy_class(h_tidy_all_slots(rv, x), x)
```
which needs to be read inside to out rather than left to right.  However, I accept that
```
rv <- h_tidy_all_slots(x)
rv <- h_tidy_class(rv, x)
```
is a reasonable compromise.

In multi-step pipes, use of the pipe makes it clear that "these steps are a single, coherent operation".  With individual  lines it's less obvious and comments may be helpful.

If you are in agreement, I would prefer to leave things as they are, simply on the basis of reducing workload.

Reply:
yeah can leave here as is just to be faster. something for next time. I am aware we don't have the contribution guideline doc for crmPack yet that defines package style which would define this